nanoFramework.DependencyInjection/DependencyInjection/ServiceProviderEngine.cs

```diff
@@ -121,6 +121,44 @@ internal object[] GetService(Type[] serviceType, IServiceCollection scopeService
             return array.Length != 0 ? array : new object[0];
         }
 
+        private ServiceDescriptor[] GetServiceDescriptors(Type serviceType, IServiceCollection scopeServices = null)
```

Comment:
It's possible this method could also be used in other locations (ie: `GetServiceObjects(Type serviceType, IServiceCollection scopeServices)`) to reduce repeated code but I figured I'd start small and just resolve the immediate issue first.